src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -251,24 +318,49 @@ public IAsyncOperation<ComputeSystemStateResult> GetState(string options)
         }).AsAsyncOperation();
     }
 
-    // Unsupported operations
-    public IAsyncOperation<ComputeSystemOperationResult> ApplyConfiguration(string configuration) => throw new NotImplementedException();
+    public IAsyncOperation<ComputeSystemThumbnailResult> GetComputeSystemThumbnailAsync(string options)
+    {
+        return Task.Run(async () =>
+        {
+            StateChanged?.Invoke(this, ComputeSystemState.Running);
+
+            var uri = new Uri(Constants.ThumbnailURI);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var randomAccessStream = await storageFile.OpenReadAsync();
 
-    public IAsyncOperation<ComputeSystemOperationResult> ApplySnapshot(string options) => throw new NotImplementedException();
+            // Convert the stream to a byte array
+            byte[] bytes = new byte[randomAccessStream.Size];
+            await randomAccessStream.ReadAsync(bytes.AsBuffer(), (uint)randomAccessStream.Size, InputStreamOptions.None);
+
+            return new ComputeSystemThumbnailResult(bytes);
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<IEnumerable<ComputeSystemProperty>> GetComputeSystemPropertiesAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            ProcessProperties();
+            return Properties;
+        }).AsAsyncOperation();
+    }
+
+    // Unsupported operations
+    public IAsyncOperation<ComputeSystemOperationResult> ApplyConfigurationAsync(string configuration) => throw new NotImplementedException();
 
-    public IAsyncOperation<ComputeSystemOperationResult> CreateSnapshot(string options) => throw new NotImplementedException();
+    public IAsyncOperation<ComputeSystemOperationResult> RevertSnapshotAsync(string options) => throw new NotImplementedException();

```

Comment:
lets update all the ones that arent implemented to 

```c#
new ComputeSystemOperationResult( new NotImplementedException(), "Method not implemented"); 
```
or something similar

Reply:
Changed.